### PR TITLE
[ResourceDetector.Azure] Follow ups

### PR DESCRIPTION
**Pull request by vishweshbankwar**
Some follow ups from https://github.com/open-telemetry/opentelemetry-dotnet-contrib/pull/1182

## Changes

Please provide a brief description of the changes here.

For significant contributions please make sure you have completed the following items:

* [ ] Appropriate `CHANGELOG.md` updated for non-trivial changes
* [ ] Design discussion issue #
* [ ] Changes in public API reviewed

### CODE REVIEW

File: src/OpenTelemetry.ResourceDetectors.Azure/AzureVmMetadataResponse.cs [hunk 1]
```diff
@@ -94,10 +94,7 @@ internal string GetValueForField(string fieldName)
                 break;
         }
 
-        if (amsValue == null)
-        {
-            amsValue = string.Empty;
-        }
+        amsValue ??= string.Empty;
```
Comment (utpilla) [inline src/OpenTelemetry.ResourceDetectors.Azure/AzureVmMetadataResponse.cs:97]: nit: Use `string amsValue = string.Empty;` at line 58 to get rid of this.

Comment (vishweshbankwar) [reply src/OpenTelemetry.ResourceDetectors.Azure/AzureVmMetadataResponse.cs:97]: It can get overwritten by one of the cases so still need this check.

Comment (utpilla) [reply src/OpenTelemetry.ResourceDetectors.Azure/AzureVmMetadataResponse.cs:97]: Ahh, these values that we are assigning could be `null`.

File: src/OpenTelemetry.ResourceDetectors.Azure/ResourceAttributeConstants.cs [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// </copyright>
+
+namespace OpenTelemetry.ResourceDetectors.Azure;
+internal class ResourceAttributeConstants
+{
+    // Appservice resource attributes
+    internal const string AppserviceSiteName = "appSrv_SiteName";
```
Comment (rajkumar-rangaraj) [inline src/OpenTelemetry.ResourceDetectors.Azure/ResourceAttributeConstants.cs:21]: Appservice -> AppService

File: src/OpenTelemetry.ResourceDetectors.Azure/ResourceAttributeConstants.cs [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+    internal const string AppserviceSlotName = "appSrv_SlotName";
+    internal const string AppserviceWsStamp = "appSrv_wsStamp";
```
Comment (rajkumar-rangaraj) [inline src/OpenTelemetry.ResourceDetectors.Azure/ResourceAttributeConstants.cs:23]: Should we drop Ws in the variable name?

File: src/OpenTelemetry.ResourceDetectors.Azure/ResourceAttributeConstants.cs [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+    internal const string AppserviceResourceGroup = "appSrv_ResourceGroup";
+
+    // Azure VM resource attributes
+    internal const string AzureVmId = "azInst_vmId";
+    internal const string AzureVmLocation = "azInst_location";
+    internal const string AzureVmName = "azInst_name";
+    internal const string AzureVmOsType = "azInst_osType";
+    internal const string AzureVmResourceGroup = "azInst_resourceGroupName";
+    internal const string AzureVmResourceId = "azInst_resourceId";
+    internal const string AzureVmsku = "azInst_sku";
```
Comment (rajkumar-rangaraj) [inline src/OpenTelemetry.ResourceDetectors.Azure/ResourceAttributeConstants.cs:35]:
```suggestion
    internal const string AzureVmSku = "azInst_sku";
```

**Review by rajkumar-rangaraj — APPROVED** [review]

**Review by utpilla — APPROVED** [review]

**Pull request merged**